packages/xo-web/src/common/xo/copy-vms-modal/index.js

```diff
@@ -7,16 +7,25 @@ import BaseComponent from 'base-component'
 import SingleLineRow from 'single-line-row'
 import Upgrade from 'xoa-upgrade'
 import { Col } from 'grid'
-import { createGetObjectsOfType } from 'selectors'
 import { SelectSr } from 'select-objects'
 import { buildTemplate, connectStore } from 'utils'
 
+import constructQueryString from '../../construct-query-string'
+import Icon from '../../icon'
+import Link from '../../link'
 import SelectCompression from '../../select-compression'
+import Tooltip from '../../tooltip'
+import { createGetObjectsOfType, createSelector } from '../../selectors'
 
 @connectStore(
   () => {
     const getVms = createGetObjectsOfType('VM').pick((_, props) => props.vms)
     return {
+      containers: createSelector(
+        createGetObjectsOfType('pool'),
+        createGetObjectsOfType('host'),
+        (pools, hosts) => ({ ...pools, ...hosts })
+      ),
       vms: getVms,
```

Comment:
```js
@connectStore({
  vms: createSelector(
    createSelector(
      createGetObjectsOfType('pool'),
      createGetObjectsOfType('host'),
        (pools, hosts) => ({ ...pools, ...hosts })
    ),
    createGetObjectsOfType('VM').pick((_, props) => props.vms),
    (containers, vms) => 
      mapValues(vms, vm => ({ 
        ...vm, 
        container: containers[vm.$container]
      })
   )
  )
})
```

Reply:
- It'll cost more resources because it'll clone  `Vms` and add an extra property to each element, IMHO, accessing the associated container when it's needed is better (in our case it will create a big collection even if the user not select the `zstd` compression).
- I'll will add complexity because we'll not know how the `container` object is injected.

Reply:
- In any case, you must access the container and this does not depend of selected compression type, right?  because you use it [here](https://github.com/vatesfr/xen-orchestra/pull/4346/files#diff-60cd5765a52975294cbb26a283612997R104).
and I think, it will also simplify  `_getVmsWithNoZstdCompression`.

-  `I'll will add complexity because we'll not know how the container object is injected.` =>  Why we don't know it ? we inject the container object, no?

Reply:
The line in the link show that it is it only used when `compression === 'zstd'` ? 

I don't see a big simplification, with this syntax `containers[vms[id].$container]` will know where the container come from but with `vm.container`, the container is injected magically in the selector.

Reply:
You are right.